src/cmd/new.rs

```diff
@@ -27,8 +27,8 @@ pub(crate) fn execute<P>(name: &str, dir: Option<P>) -> Result<Option<String>>
 where
     P: AsRef<Path>,
 {
-    if name.contains('-') {
-        anyhow::bail!("Contract names cannot contain hyphens");
+    if !name.replace('_', "").chars().all(char::is_alphanumeric) {
```

Comment:
```suggestion
    if !name.chars().all(|c| c.is_alphanumeric() || c == '_') {
```

I think something like this is more readable (note I haven't checked whether that compiles or not)

Reply:
Good idea, probably a tad more efficient too. Made the change and it does indeed compile